terraform/github-to-gitlab-mirror/main.tf

```diff
@@ -0,0 +1,56 @@
+locals {
+  user = "nix-community"
+  blacklist = [
+    # invalid name
+    ".github",
+    # do already exists
+    "emacs-overlay",
+    "docker-nixpkgs",
+    "nixos-generators",
+    "poetry2nix",
+  ]
+  # remove blacklist
+  filteredRepos = [
+    for repo in data.github_repositories.repos.full_names : repo if !contains(local.blacklist, element(split("/", repo), 1))
+  ]
+}
+
+data "github_repositories" "repos" {
+  query = "user:${local.user} is:public"
+}
+
+data "github_repositories" "non-archived-repos" {
+  query = "user:${local.user} is:public archived:false"
+}
+
+resource "gitlab_group" "group" {
+  name             = local.user
+  path             = local.user
+  visibility_level = "public"
+}
+
+resource "gitlab_project" "repos" {
+  for_each               = toset(local.filteredRepos)
+  name                   = element(split("/", each.key), 1)
+  namespace_id           = gitlab_group.group.id
+  import_url             = "https://github.com/${each.key}"
+  shared_runners_enabled = false
+  visibility_level       = "public"
+}
+
+# when the api, we run into api rate limits
```

Comment:
Does this mean we cann't use webhooks for this?

Reply:
Clarified the gibberish I wrote yesterday. Pull mirroring is not available to free gitlab accounts unless you enabled the setting before they made it a premium feature.